src/UI/Components/Diagram/Canvas.tsx

```diff
@@ -1,435 +1,314 @@
-
-      switch (action) {
-        case "update":
-          newInstance.action =
-            updatedInstance?.action === "create" ? "create" : "update";
-
-          return new Map(prevInstances.set(cell.id as string, newInstance));
-        case "create":
-          newInstance.action = action;
-
-          return new Map(prevInstances.set(cell.id as string, newInstance));
-        default:
-          if (
-            updatedInstance &&
-            (updatedInstance.action === null ||
-              updatedInstance.action === "update")
-          ) {
-            return new Map(
-              prevInstances.set(cell.id as string, {
-                instance_id: cell.id,
-                service_entity: cell.getName(),
-                config: {},
-                action: "delete",
-                attributes: null,
-                edits: null,
-                embeddedTo: cell.attributes.embeddedTo,
-                relatedTo: cell.attributes.relatedTo,
-              }),
-            );
-          } else {
-            const newInstances = new Map(prevInstances);
-
-            newInstances.delete(cell.id as string);
-
-            return newInstances;
-          }
-      }
-    });
-  };
-
-  useEffect(() => {
-    const connectionRules = createConnectionRules(services, {});
+    const connectionRules = createConnectionRules(serviceModels, {});
     const actions = diagramInit(
-      canvas,
+      Canvas,
+      (newScroller) => {
+        setScroller(newScroller);
+      },
       connectionRules,
-      handleUpdate,
       editable,
+      mainService,
     );
 
     setDiagramHandlers(actions);
-    if (instance) {
-      const isMainInstance = true;
-
-      try {
-        const cells = actions.addInstance(instance, services, isMainInstance);
-        const newInstances = new Map();
-
-        cells.forEach((cell) => {
-          if (cell.type === "app.ServiceEntityBlock") {
-            newInstances.set(cell.id, {
-              instance_id: cell.id,
-              service_entity: cell.entityName,
-              config: {},
-              action: null,
-              attributes: cell.instanceAttributes,
-              embeddedTo: cell.embeddedTo,
-              relatedTo: cell.relatedTo,
-            });
-          }
-        });
-        setInstancesToSend(newInstances);
-      } catch (error) {
-        setAlertType(AlertVariant.danger);
-        setAlertMessage(String(error));
-      }
-    }
 
     return () => {
+      setStencilState(createStencilState(mainService));
+      setIsStencilStateReady(false);
       actions.removeCanvas();
     };
-  }, [instance, services, mainServiceName, editable]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [mainService, serviceModels, isStencilStateReady]);
 
+  /**
+   * create the stencil sidebar only if sidebar ref, the scroller, related inventories by inter-service relations, main service and service models are defined
```

Comment:
your comment still talks about stencil sidebar
```suggestion
   * create the left-sidebar only if the left-sidebar ref, the scroller, the related inventories by inter-service relations, the main service and service models are defined
```